backend/site-api/src/app/site/get_site.rs

```diff
@@ -78,11 +94,32 @@ pub async fn get_site(
                 site_metadata.site_type,
             )
             .await;
-        return Ok(Json(to_api_response(
-            site,
-            admin_or_owner,
-            site_metadata.disabled,
-        )));
+
+        if let Some(update_key) = query.update_key {
+            return DateTime::parse_from_rfc3339(update_key.as_str())
+                .map_err(|e| {
+                    ApiError::internal_error().message(format!("Date parsing error: {}", e))
+                })
+                .and_then(|date_time| {
+                    if site.updated_at == date_time {
+                        Ok(StatusCode::NO_CONTENT.into_response())
```

Comment:
I adjusted the endpoints to return `204` response instead of empty data. The reason is that it's a little simpler for the frontend to deal with, and slightly easier to maintain in the backend, if site response fields change. I had to make this `get_site_result` helper function in order to return `Result<Response, ApiError>`, since Axum won't allow that to be returned from the top level handler.